### PR TITLE
Adds missing meta data to Entry, Asset and ContentType responses

### LINKED ISSUE #44: Contentful.SysData missing important information
In previous versions, the map that has been replaced with `Contentful.SysData` contained a number of important fields. These included:

```elixir
...
  sys: %{
    contentType: %{
      sys: %{
        id: "<content-type-from-model>",  # for example: "author"
        ...
      }
    },
    createdAt: "<created-timestamp>",
    updatedAt: "<updated-timestamp>",
    locale: "<locale-string>",  # eg: "en-GB",
    ...
  }
...
```

I'd like to update to the latest Contentful version, but my system is reliant on these fields.

Can these fields be returned to the `Contentful.Entry`/`Contentful.Asset` or `Contentful.SysData` structs?

### PR DESCRIPTION
**Pull request by floriank**
This will add missing meta information back into the resolved responses:

## Assets

```elixir
import Contentful.Query
alias Contentful.Asset
alias Contentful.Delivery.Assets

{:ok, asset} = Assets |> fetch_one("my_asset_id")

"my_asset_id" = asset.id
"<a timestamp for updated_at>" = asset.sys.updated_at
"<a timestamp for created_at>" = asset.sys.created_at
"<a locale string>" = asset.sys.locale
```

## Entries

```elixir
import Contentful.Query
alias Contentful.{ContentType, Entry, SysData}
alias Contentful.Delivery.Entries

{:ok, entry} = Entries |> fetch_one("my_entry_id")

"my_entry_id" = entry.id
"<a timestamp for updated_at>" = entry.sys.updated_at
"<a timestamp for created_at>" = entry.sys.created_at
"<a locale string>" = entry.sys.locale
%ContentType{id: "the_associated_content_type_id"} =  entry.sys.content_type
```

## ContentTypes
```elixir
import Contentful.Query
alias Contentful.ContentType
alias Contentful.Delivery.ContentTypes

{:ok, content_type} = ContentTypes |> fetch_one("my_content_type_id")

"my_content_type_id" = content_type.id
"<a timestamp for updated_at>" = content_type.sys.updated_at
"<a timestamp for created_at>" = content_type.sys.created_at
```

## Why is the `id` not part of `sys`?

Because I am lazy and the id is most likely just important enough to warrant being access by `entity.id` instead of `entity.sys.id`. I am open to feedback here :wink: 

Shoutout to @OldhamMade for bringing this to my attention in #44.

fixes #44.

### CODE REVIEW

**Comment by asaaki** [comment]
> Why is the `id` not part of `sys`?

When I was working on a similar (more hobby) project/SDK for another language I also struggled with the `sys` property. In the end I think it is fine to not have structural equality between API presentation and language environment, the only important thing is that the necessary information there. Esp. for the CDA, which is a read-only API, we can flatten or strip presentational structure if it makes handling the data more convenient, since we don't really need to keep it for updates (a CMA concern).

tl;dr: I'm totally fine with structural changes for a "one-way" API.

File: lib/contentful/sys_data.ex [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
-  defstruct [:id, :revision, :version]
+  defstruct [:id, :revision, :version, :created_at, :updated_at, locale: nil, content_type: nil]
 
   @type t :: %Contentful.SysData{
           id: String.t(),
           revision: integer() | nil,
-          version: integer() | nil
+          version: integer() | nil,
+          created_at: String.t() | nil,
+          updated_at: String.t() | nil,
```
Comment (asaaki) [inline lib/contentful/sys_data.ex:17]: Not sure about this one, might be optionally present I guess.

File: lib/contentful/sys_data.ex [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
+          # NOTE: locale string only exists in entries and assets
+          locale: String | nil,
+          # NOTE: ContentType only exists for entries
+          content_type: %ContentType{} | nil
```
Comment (asaaki) [inline lib/contentful/sys_data.ex:21]: Here I wish we had some kind of dependent typing (if the thing is an entry the content type must present).
A less pragmatic way would be to have a sys data enum with specific variants for the different data types; I'm currently experimenting with that idea in a hobby SDK.

**Comment by floriank** [comment]
Ultimately, there probably will be a CMA integration with this library - that's when the question of `sys.id` or `id` will probably pop up again, but it should be easy to serialize/deserialize this on the fly.

Don't quote me years from now please.

**Review by asaaki — APPROVED** [review]
Looks good to me. 👍

File: lib/contentful/sys_data.ex (hunk 2 above)
Comment (floriank) [reply lib/contentful/sys_data.ex:21]: I am not completely sure, if we really should wrap this into a `ContentType` here. Ultimately I opted for consistency, since it'll give you a possibility to match this at run time.

The `ContentType` isn't complete though, since it gets manually created from a `"Link"`.

I am still looking into how to generically allow for resolving links from arbitrary entities.

File: lib/contentful/sys_data.ex (hunk 1 above)
Comment (floriank) [reply lib/contentful/sys_data.ex:17]: According to https://www.contentful.com/developers/docs/references/content-delivery-api/#/introduction/common-resource-attributes, this should be present for `Entry`, `Asset` and `ContentTypes`.

File: lib/contentful/sys_data.ex (hunk 2 above)
Comment (floriank) [reply lib/contentful/sys_data.ex:21]: I'd actually revert this to just be an optional `String.t()` I guess

File: lib/contentful/sys_data.ex (hunk 1 above)
Comment (floriank) [reply lib/contentful/sys_data.ex:17]: That also explicitly excludes `Space`, it seems.
